src/gh/diffCheck/diffCheck/df_geometries.py

```diff
@@ -297,22 +299,25 @@ def __post_init__(self):
         self._index_assembly = None
 
         self._center = None
+        self.__id = uuid.uuid4().int
 
     def deepcopy(self):
         return DFBeam(self.name, [face.deepcopy() for face in self.faces])
 
+
     @classmethod
-    def from_brep_face(cls, brep):
+    def from_brep_face(cls, brep, is_cylinder=False):
```

Comment:
Actually I don't see how to avoid this.
`from_brep_face()` is a constructor of DFBeam and the way it is used until now makes it difficult to avoid passing the "is_roundwood" as parameter in the constructor. 

For example DFBuildAssembly creates the beams like this: 

```
beam = DFBeam.from_brep_face(brep, i_is_roundwood)
beams.append(beam)
```
and from_brep_face() triggers immediately the joint detector, for which we need to know if it is a roundwood beam or not:
(df_geometrie.py, line 304)
```
def from_brep_face(cls, brep, is_roundwood=False):
     """
     Create a DFBeam from a RhinoBrep object.
     It also removes duplicates and creates a list of unique faces.
     """
     faces : typing.List[DFFace] = []
     data_faces = diffCheck.df_joint_detector.JointDetector(brep, is_roundwood).run()
```
So in order to have this info at the joint detector stage, I don't see how we can avoid passing it in the constructor.

Let me know if i'm missing something !